src/unknown/component.rs

```diff
@@ -0,0 +1,29 @@
+// 
+// Sysinfo
+// 
+// Copyright (c) 2018 Guillaume Gomez
+//
+
+use ComponentExt;
+
+/// Dummy struct representing a component.
+pub struct Component {
```

Comment:
The parens are useless in this case. Any reason for why you added them?

Reply:
So that one cannot create such a struct.

Reply:
Well, you can:

```rust
pub struct Foo {}

fn main() {
    let x = Foo {};
}
```

Or did I miss anything?

Reply:
Not from an external crate. That's the different with `pub struct Foo;` where you can.

Reply:
Interesting, didn't know that. Thanks for the info!

Reply:
I know you’re going to find that weird but I just read that comment.

And @GuillaumeGomez was right, you can still create the `struct`, even from an external crate. I guess @tomaka confused with `enum Foo {}`, which cannot be constructed by anyone.

To prevent someone else to create your `struct`:

```rust
struct Foo {
  _hidden: ()
}
```